src/mscorlib/src/System/IO/FileLoadException.cs

```diff
@@ -2,182 +2,113 @@
 
-            if (_fileName != null && _fileName.Length != 0)
-                s += Environment.NewLine + SR.Format(SR.IO_FileName_Name, _fileName);
+            if (FileName != null && FileName.Length != 0)
+                s += Environment.NewLine + SR.Format(SR.IO_FileName_Name, FileName);
 
             if (InnerException != null)
                 s = s + " ---> " + InnerException.ToString();
 
             if (StackTrace != null)
                 s += Environment.NewLine + StackTrace;
 
-            try
-            {
-                if (FusionLog != null)
-                {
-                    if (s == null)
-                        s = " ";
-                    s += Environment.NewLine;
-                    s += Environment.NewLine;
-                    s += FusionLog;
-                }
-            }
-            catch (SecurityException)
+            if (FusionLog != null)
             {
+                if (s == null)
+                    s = " ";
+                s += Environment.NewLine;
+                s += Environment.NewLine;
+                s += FusionLog;
             }
 
             return s;
         }
 
-        protected FileLoadException(SerializationInfo info, StreamingContext context) : base(info, context)
+        protected FileLoadException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
         {
             // Base class constructor will check info != null.
 
-            _fileName = info.GetString("FileLoad_FileName");
+            FileName = info.GetString("FileLoad_FileName");
 
             try
             {
-                _fusionLog = info.GetString("FileLoad_FusionLog");
+                FusionLog = info.GetString("FileLoad_FusionLog");
             }
             catch
             {
-                _fusionLog = null;
+                FusionLog = null;
             }
         }
 
-        private FileLoadException(String fileName, String fusionLog, int hResult)
-            : base(null)
-        {
-            SetErrorCode(hResult);
-            _fileName = fileName;
-            _fusionLog = fusionLog;
-            SetMessageField();
-        }
-
-        public String FusionLog
-        {
-            get { return _fusionLog; }
-        }
-
         public override void GetObjectData(SerializationInfo info, StreamingContext context)
         {
             // Serialize data for our base classes.  base will verify info != null.
             base.GetObjectData(info, context);
 
             // Serialize data for this class
-            info.AddValue("FileLoad_FileName", _fileName, typeof(String));
-
-            try
-            {
-                info.AddValue("FileLoad_FusionLog", FusionLog, typeof(String));
-            }
-            catch (SecurityException)
+            info.AddValue("FileLoad_FileName", FileName, typeof(string));
+            string fusionLog = FusionLog;
+            if (fusionLog != null)
```

Comment:
Hmm... I think I unified the "missing because of security" to "missing because of non-null or security" in the original code. Fixed.